components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/OpenIDConnectSystemClaimImpl.java

```diff
@@ -145,8 +145,15 @@ private boolean isCodeHashApplicable(String responseType) {
 
     private boolean isAccessTokenHashApplicable(String responseType) {
         // At_hash is generated on an access token. Therefore check whether the response type returns an access_token.
-        // id_token and none response types don't return and access token
-        return !OAuthConstants.ID_TOKEN.equalsIgnoreCase(responseType) &&
-                !OAuthConstants.NONE.equalsIgnoreCase(responseType);
+        for (String respType : getResponseTypes(responseType)) {
+            if (OAuthConstants.TOKEN.equalsIgnoreCase(respType)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private String[] getResponseTypes(String responseType) {
```

Comment:
formatting issue. should have a new line after method signature. Add the code style to your IDE

Reply:
can we do a null and blank check here? StringUtils.isNotBlank(responseType) and add a debug log if it is blank.